pkg/kube/services/services.go

```diff
@@ -149,39 +149,46 @@ func FindIngressURL(client kubernetes.Interface, namespace string, name string)
 
 // IngressURL returns the URL for the ingres
 func IngressURL(ing *nv1.Ingress) string {
-	if ing != nil {
-		if len(ing.Spec.Rules) > 0 {
-			rule := ing.Spec.Rules[0]
-			hostname := rule.Host
-			for _, tls := range ing.Spec.TLS {
-				for _, h := range tls.Hosts {
-					if h != "" {
-						url := "https://" + h
-						log.Logger().Debugf("found service url %s", url)
-						return url
-					}
-				}
-			}
-			ann := ing.Annotations
-			if hostname == "" && ann != nil {
-				hostname = ann[kube.AnnotationHost]
-			}
-			if hostname != "" {
-				url := "http://" + hostname
-				if rule.HTTP != nil {
-					if len(rule.HTTP.Paths) > 0 {
-						p := rule.HTTP.Paths[0].Path
-						if p != "" {
-							url += p
-						}
-					}
-				}
+	if ing == nil {
```

Comment:
Always good to return early, instead of having multiple branches, it's confusing and increases complexity.